pulsar/consumer_partition.go

```diff
@@ -597,7 +602,18 @@ func (pc *partitionConsumer) MessageReceived(response *pb.CommandMessage, header
 			pc.AckID(msgID)
 			continue
 		}
-
+		var messageIndex *uint64
+		var brokerPublishTime *time.Time
+		if brokerMetadata != nil {
```

Comment:
It looks like we've checked above that the brokerMetadata object is nil by doing the following:

```
brokerMetadata, err := reader.ReadBrokerMetadata()
	if err != nil { // here
		// todo optimize use more appropriate error codes
		pc.discardCorruptedMessage(pbMsgID, pb.CommandAck_BatchDeSerializeError)
		return err
	}
```

Do we still need to double check here? Will someone concurrently modify the value of the brokerMetadata object?

Reply:
`brokermetadata` can be null if broker not enable this feature